### PR TITLE
Automated cherry pick of #73309: Should move all unscheduable pods when we received move request to active queue

### PR DESCRIPTION
**Pull request by cofyc**
Cherry pick of #73309 on release-1.12.

#73309: Fix weakness of current receivedMoveRequest

### CODE REVIEW

**Comment by cofyc** [comment]
/assign @feiskyer
for patch release

File: pkg/scheduler/core/scheduling_queue.go [hunk 1]
```diff
@@ -191,12 +203,9 @@ type PriorityQueue struct {
...
 	nominatedPods *nominatedPodMap
-	// receivedMoveRequest is set to true whenever we receive a request to move a
-	// pod from the unschedulableQ to the activeQ, and is set to false, when we pop
-	// a pod from the activeQ. It indicates if we received a move request when a
-	// pod was in flight (we were trying to schedule it). In such a case, we put
-	// the pod back into the activeQ if it is determined unschedulable.
-	receivedMoveRequest bool
+	// schedulingCycle represents sequence number of scheduling cycle and is incremented
+	// when a pod is popped.
+	schedulingCycle int64
```
Comment (Huang-Wei) [inline pkg/scheduler/core/scheduling_queue.go]: missing `moveRequestCycle` here

Comment (cofyc) [reply pkg/scheduler/core/scheduling_queue.go]: fixed, thanks!

**Review by Huang-Wei — APPROVED** [review]
/lgtm

/assign @bsalamat 
for approval

**Review by bsalamat — COMMENTED** [review]
/approve

**Pull request merged**